templates/lxc-download.in

```diff
@@ -147,7 +147,17 @@ gpg_setup() {
   done
 
   if [ -z "${success}" ]; then
-    echo "ERROR: Unable to fetch GPG key from keyserver"
+    {
+      echo "ERROR: Unable to fetch GPG key from keyserver"
+      TMP="$(basename "${DOWNLOAD_KEYSERVER}")"
+      if which dig > /dev/null \
+        && [ "$(dig "$TMP" +short)" = "" ] ; then
+        echo "$TMP is not a valid domain"
+      fi
+      echo "try overriding the default keyserver with"
+      echo "export DOWNLOAD_KEYSERVER=\"hkp://keyserver.ubuntu.com\""
+      echo "or updating lxc from https://github.com/lxc/lxc#building-lxc"
```

Comment:
Would something like this work instead:
```
if ! getent hosts "$(basename "${DOWNLOAD_KEYSERVER}")"; then
    echo "Unable to resolve GPG keyserver at: ${DOWNLOAD_KEYSERVER}"
    echo "This may indicate a network issue on your system or a problem with the GPG keyserver"
    echo "You can override the keyserver using: export DOWNLOAD_KEYSERVER=hkp://SOME-SERVER"

fi
```

Reply:
The current keyserver in this very file is `keyserver.ubuntu.com` so it makes little sense to tell people to override it to the same value :)

I'd also like to stay away from telling people to go read source code on Github. Most should really file a bug against their distro so they can get an updated package instead :)

Reply:
`getent` is likely? more likely available than `dig` so that's good.

`> /dev/null` should stay because echoing unexplained output may confuse users. 

There should be an error message if the domain is bad and also if it's not.

Using `hkp://SOME-SERVER` seems fine because `export DOWNLOAD_KEYSERVER=` is the helpful bit.

Technically the link is to the "building"/installing instructions not to to "go read source code" and I think that's important because Ubuntu is possibly the least effective place to submit a bug report or patch, so installing from upstream is the only effective workaround (YMMV but my experience is they simply don't have the human power or in depth knowledge of all the packages).

Reply:
I'm the maintainer for the LXC package in Ubuntu ;)

My worry with telling them to follow the build instructions is that it's actually been causing us a lot of issues.
Most people who try to go the build from source route will forget to also purge all LXC packages from their systems and so end up with conflicting files all over, their PATH may be selecting the newer versions of the binaries but their ld config will load the old library, causing all kind of crashes and weird errors which can take us days to figure out when they file an issue.

In comparison, them putting a simple `export` in their bashrc or something, that's a lot easier to deal with.

And yeah, it's not been a top priority of ours lately because of the focus on LXD which we can rollout a fix for in less than a couple of hours whereas the SRU process in Ubuntu can be quite a bit of paperwork and weeks of delays...

Reply:
Maybe the upstream comment could only be shown if github.com resolves but the key server does not.
That would prevent most people with network issues from even seeing the upstream error/help message.

I'm not picky, and your the maintainer so whatever you think is best.